tensorflow_addons/layers/normalizations.py

```diff
@@ -455,35 +454,42 @@ def get_config(self):
         return dict(**base_config, **config)
 
     def _create_input_spec(self, input_shape):
-        ndims = len(tf.TensorShape(input_shape))
+        ndims = len(input_shape)
+
         for idx, x in enumerate(self.axis):
             if x < 0:
                 self.axis[idx] = ndims + x
 
-        # Validate axes
         for x in self.axis:
-            if x < 0 or x >= ndims:
+            if x < 0 or x >= len(input_shape):
                 raise ValueError("Invalid axis: %d" % x)
 
-        if len(self.axis) != len(set(self.axis)):
-            raise ValueError("Duplicate axis: %s" % self.axis)
+            elif x == 0:
+                raise ValueError(
+                    "You are trying to normalize your batch axis. "
+                    "Use tf.layers.batch_normalization instead."
+                )
 
         axis_to_dim = {x: input_shape[x] for x in self.axis}
         self.input_spec = tf.keras.layers.InputSpec(ndim=ndims, axes=axis_to_dim)
 
     def _check_axis(self, axis):
-        if not isinstance(axis, list):
+        if isinstance(axis, list):
+            self.axis = axis
+
+        elif isinstance(axis, int):
+            if axis != -1 or self.channel_idx != -1:
+                raise ValueError("Expected index for 2D is -1 but got {}".format(axis))
+
+            self.axis = [axis]
+
+        else:
             raise TypeError(
                 """Expected a list of values but got {}.""".format(type(axis))
```

Comment:
Expected a *list of values or int*...

Reply:
Will do, also do you require type hint for axis to be changed to `Union[list, int]`?

Reply:
Yup.